baker/GrapherBaker.tsx

```diff
@@ -43,56 +45,86 @@ import {
-            isPreviewing
-            // todo: we're not ready to publish datapages yet, so we only want to
-            // render them if we're previewing.
-            // (datapage.status === "published" || isPreviewing)
-        ) {
+
+    // Get the datapage JSON file from the owid-content git repo that has
+    // been updated and pulled separately by an author.
+    const { datapageJson, parseErrors } = await getDatapageJson(id)
+
+    // When previewing a datapage we want to show all discrepancies with the
+    // expected JSON schema in the browser. When baking, a single error by
+    // datapage will be sent to slack while falling back to rendering a regular
+    // grapher page.
+    if (parseErrors.length > 0) {
+        if (isPreviewing) {
             return renderToHtmlPage(
-                <DataPage
-                    grapher={grapher}
-                    datapage={datapage}
-                    baseUrl={BAKED_BASE_URL}
-                    baseGrapherUrl={BAKED_GRAPHER_URL}
-                />
+                <pre>{JSON.stringify(parseErrors, null, 2)}</pre>
             )
-        }
-    } catch (e: any) {
-        // Do not throw an error if the datapage JSON does not exist, but rather
-        // if it does and it fails to parse or render.
-        if (e.code !== "ENOENT") {
-            logErrorAndMaybeSendToSlack(
-                `Failed to render datapage ${fullPath}. Error: ${e}`
+        } else {
+            logContentErrorAndMaybeSendToSlack(
+                new JsonError(
+                    `Data page error in ${id}.json: please check ${ADMIN_BASE_URL}/admin/grapher/${grapher.slug}`
+                )
             )
         }
     }
-    // fallback to regular grapher page
-    return renderGrapherPage(grapher)
+
+    // Fallback to rendering a regular grapher page whether the datapage JSON
+    // wasn't found or failed to parse or if the datapage is not fully
+    // configured for publishing yet
+    if (
+        // This could be folded into the showDataPageOnChartIds check below, but
+        // is kept separate to reiterate that that abscence of a datapageJson leads to
+        // rendering a grapher page
+        !datapageJson ||
+        parseErrors.length > 0 ||
+        // We only want to render datapages on selected charts, even if the
+        // variable found on the chart has a datapage configuration.
+        !grapher.id ||
+        !datapageJson.showDataPageOnChartIds.includes(grapher.id) ||
+        // Fall back to rendering a regular grapher page if the datapage is not
+        // published or if we're not previewing
+        !(datapageJson.status === "published" || isPreviewing)
+    )
+        return renderGrapherPage(grapher)
+
+    // Compliment the text-only content from the JSON with rich text from the
+    // companion gdoc
+    const datapageGdoc = await getDatapageGdoc(
+        datapageJson,
+        isPreviewing,
+        publishedExplorersBySlug
```

Comment:
For the moment (and I'll be the first to admit this isn't great) we only need `publishedExplorersBySlug` when rendering prominent links to explorers downstream. 

If that's not a thing you anticipate needing to do then you can just pass `{}`

And soon, once the DB has the explorer data then we won't have to pass this object around at all.

Reply:
Hum difficult to know for sure. We've already seen full blown explorers being included in data pages where we only anticipated grapher charts, so I wouldn't say prominent links are too far out.

Good to know though, thanks for the pointers.